### PR TITLE
feat(wallet): add wallet service API

### LINKED ISSUE #1210: Wallet Service APIs
Here is the list of APIs that wallet service should provide

- [x] Create Wallet  - Create a wallet and return the mnemonic
- [x]  Restore Wallet - Recover wallet by given mnemonic
- [x]  Load Wallet - Load wallet by name
- [x]  Unload Wallet - Unload wallet by name
- [x]  Get New Address - Get new address based on the given address type
- [x]  Get Total Balance - Get the total balance of the wallet
- [ ]  Get Total Stake - Get the total stake of the wallet
- [ ] Get Address Info - Get the address info for the given address
- [ ] Set Label - Set label for the given address
- [ ] Get Wallet Info - Return the information about wallet
- [ ] List Wallets - Return the list of wallets inside wallet directory
- [ ] List Address - Return the list of addresses in wallet, it accepts Purpose as argument

### LINKED ISSUE #1342: GetAddressHistory is missed from Swagger API
GetAddressHistory API should add to the Swagger API.

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

- [x]  Get Total Stake - Get the total stake of the wallet
- [x] Get Address Info - Get the address info for the given address
- [x] Set Label - Set label for the given address
- [x] Get Wallet Info - Return the information about wallet
- [x] List Wallets - Return the list of wallets inside wallet directory
- [x] List Address - Return the list of addresses in wallet, it accepts 

## Related issue(s)

- Resolves #1210
- Resolves #1342

### CODE REVIEW

**Comment by Ja7ad** [comment]
@b00f Can write more details what is it response of:

- Get Wallet Info
- List Wallets
- List Address

**Comment by b00f** [comment]
> Get Wallet Info

Information about wallet, like Wallet name, version, Is Encrypted, Creation time, etc

> List Wallets

For listing wallets, we can list files inside the `wallets` directory. The better solution is to keep the list of wallets in `wallets.json`. 

> List Address

The list of all addresses in a wallet.

File: wallet/manager.go [hunk 1]
```diff
@@ -243,3 +245,51 @@ func (wm *Manager) SetAddressLabel(walletName, address, label string) error {
...
+		wallets = append(wallets, filepath.Base(f))
+	}
+
+	return wallets, nil
+}
+
+func (wm *Manager) ListAddress(walletName string) ([]vault.AddressInfo, error) {
+	wlt, ok := wm.wallets[walletName]
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "wallet is not loaded")
```
Comment (b00f) [inline wallet/manager.go:291]: One potential refactor is to define proper error types here.

File: wallet/manager.go [hunk 2]
```diff
@@ -243,3 +245,51 @@ func (wm *Manager) SetAddressLabel(walletName, address, label string) error {
...
+	}
+
+	return append(wlt.AllValidatorAddresses(), wlt.AllAccountAddresses()...), nil
```
Comment (b00f) [inline wallet/manager.go:294]: Use `wlt.AddressInfos()`

File: wallet/wallet.go [hunk 3]
```diff
@@ -30,6 +30,16 @@ type Wallet struct {
 	grpcClient *grpcClient
 }
 
+type Info struct {
+	WalletName string
+	Version    int64
+	Network    string
+	UUID       string
+	Encrypted  bool
+	Crc        uint32
```
Comment (b00f) [inline wallet/wallet.go:39]: Do we really need to return Crc? I don't see any reason for that

**Review by b00f — CHANGES_REQUESTED** [review]

File: wallet/manager.go (hunk 1 above)
Comment (b00f) [reply wallet/manager.go:283]: We should define error types here.

**Pull request merged**